util/util.go

```diff
@@ -591,3 +592,15 @@ func LowestNonZeroResult(i, j ctrl.Result) ctrl.Result {
 		return j
 	}
 }
+
+// IsNil returns an error if the passed interface is equal to nil or if it has an interface value of nil.
+func IsNil(i interface{}) bool {
+	if i == nil {
+		return true
+	}
+	switch reflect.TypeOf(i).Kind() {
+	case reflect.Ptr, reflect.Map, reflect.Array, reflect.Chan, reflect.Slice, reflect.Interface:
```

Comment:
Added reflect.Interface here - though the tests weren't failing when it wasn't present. My working theory is that a nil interface is always a pointer to nil - but I'm not sure if that's the correct interpretation. 

In the debugger those cases were always classified as type = Ptr.

Reply:
I'm not sure if reflect.Interface is possible to be honest

Reply:
Hm should we add UnsafePointer and Func as well?

xref: https://github.com/golang/go/blob/master/src/reflect/value.go#L1537-L1555

Reply:
Looks like then we have all kind's for which .IsNil doesn't panic